File: ComplementApp.API/Dtos/FormatoSolicitudPagoDto.cs
using System;
using System.Collections.Generic;
using ComplementApp.API.Models;

namespace ComplementApp.API.Dtos
{
    public class FormatoSolicitudPagoDto
    {
        public int FormatoSolicitudPagoId { get; set; }
        public int PlanPagoId { get; set; }
        public DateTime FechaSistema { get; set; }
        public DateTime FechaInicio { get; set; }
        public DateTime FechaFinal { get; set; }
        public decimal ValorFacturado { get; set; }
        public int MesId { get; set; }
        public int EstadoId { get; set; }
        public string MesDescripcion { get; set; }
        public string NumeroPlanilla { get; set; }
        public string NumeroFactura { get; set; }
        public string Observaciones { get; set; }
        public decimal ValorBaseGravableRenta { get; set; }
        public decimal ValorIva { get; set; }
        public decimal BaseCotizacion { get; set; }
        public decimal ValorPagadoFechaActual { get; set; }
        public string ObservacionesModificacion { get; set; }
        public decimal AporteSalud { get; set; }
        public decimal AportePension { get; set; }
        public decimal RiesgoLaboral { get; set; }
        public decimal FondoSolidaridad { get; set; }
        public int NumeroPagoFechaActual { get; set; }
        public int CantidadMaxima { get; set; }
        public string NumeroRadicadoProveedor { get; set; }
        public DateTime FechaRadicadoProveedor { get; set; }
        public string NumeroRadicadoSupervisor { get; set; }
        public string ModalidadContrato { get; set; }
        public string TipoAdminPila { get; set; }
        public DateTime FechaRadicadoSupervisor { get; set; }        
        public CDPDto Cdp { get; set; }
        public TerceroDto Tercero { get; set; }
        public ContratoDto Contrato { get; set; }
        public PlanPagoDto PlanPago { get; set; }
        public ValorSeleccion ActividadEconomica { get; set; }
        public ICollection<CDPDto> PagosRealizados { get; set; }
        public ICollection<DetalleFormatoSolicitudPagoDto> detallesFormatoSolicitudPago { get; set; }

    }
}

File: ComplementApp-SPA/src/app/_dto/formatoSolicitudPagoDto.ts
import { Cdp } from '../_models/cdp';
import { Contrato } from '../_models/contrato';
import { PlanPago } from '../_models/planPago';
import { Tercero } from '../_models/tercero';
import { DetalleFormatoSolicitudPagoDto } from './detalleFormatoSolicitudPagoDto';
import { ValorSeleccion } from './valorSeleccion';

export interface FormatoSolicitudPagoDto {
  formatoSolicitudPagoId: number;
  planPagoId: number;
  fechaInicio: Date;
  fechaFinal: Date;
  fechaSistema: Date;
  valorFacturado: number;
  mesId: number;
  mesDescripcion: string;
  numeroPlanilla: string;
  numeroFactura: string;
  observaciones: string;
  valorBaseGravableRenta: number;
  valorIva: number;
  baseCotizacion: number;
  observacionesModificacion: string;

  valorPagadoFechaActual: number;
  numeroPagoFechaActual: number;
  cantidadMaxima: number;
  aporteSalud: number;
  aportePension: number;
  riesgoLaboral: number;
  fondoSolidaridad: number;
  estadoId: number;
  numeroRadicadoProveedor: string;
  fechaRadicadoProveedor: Date;
  numeroRadicadoSupervisor: string;
  fechaRadicadoSupervisor: Date;
  tipoAdminPila: string;

  actividadEconomica: ValorSeleccion;
  planPago: PlanPago;
  pagosRealizados: Cdp[];
  cdp: Cdp;
  tercero: Tercero;
  contrato: Contrato;
  detallesFormatoSolicitudPago: DetalleFormatoSolicitudPagoDto[];
}
